src/main/java/org/wso2/carbon/mongodb/util/MongoDatabaseUtil.java

```diff
@@ -117,6 +117,27 @@ public static DB createRealmDataSource(RealmConfiguration realmConfiguration) th
                     "' provided in user_mgt.xml does not contain the database name. Cannot start server!");
         }
 
+        MongoClient mongoClient = null;
+        synchronized (MongoDatabaseUtil.class) {
+            if (mongoClients == null) {
+                mongoClients = new HashMap<>();
+            } else {
+                mongoClient = mongoClients.get(urlWithCredentials);
+            }
+            if (mongoClient == null) {
+                mongoClient = new MongoClient(clientURI);
+                mongoClients.put(urlWithCredentials, mongoClient);
+
+                if(log.isDebugEnabled()) {
+                    log.debug("create new mongoClient");
```

Comment:
better change the log message "Created new mongoClient with username: " +userName